jormungandr-watchdog/examples/stdin_echo.rs

```diff
@@ -83,20 +90,109 @@ impl Service for StdoutWriter {
     }
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum LogLevel {
+    INFO,
+    WARN,
+    DEBUG,
+    ERROR,
+    TRACE,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct LoggerConfig {
+    level: LogLevel,
+}
+
+impl Default for LoggerConfig {
+    fn default() -> Self {
+        LoggerConfig {
+            level: LogLevel::ERROR,
+        }
+    }
+}
+
+fn log_level_into_tracing_level(level: &LogLevel) -> tracing::Level {
+    return match level {
+        LogLevel::INFO => tracing::Level::INFO,
+        LogLevel::WARN => tracing::Level::WARN,
+        LogLevel::DEBUG => tracing::Level::DEBUG,
+        LogLevel::ERROR => tracing::Level::ERROR,
+        LogLevel::TRACE => tracing::Level::TRACE,
+    };
+}
+
+impl Settings for LoggerConfig {
+    fn add_cli_args<'a, 'b>() -> Vec<Arg<'a, 'b>> {
+        vec![Arg::with_name("Log level")
+            .short("ll")
+            .long("log_level")
+            .takes_value(true)
+            .default_value("Warning")
+            .value_name("LOG_LEVEL")
+            .help("Services log level: []")]
```

Comment:
add the `possible_values`: https://docs.rs/clap/2.33.0/clap/struct.Arg.html#method.possible_values

Reply:
also allow for `env` : https://docs.rs/clap/2.33.0/clap/struct.Arg.html#method.env

Reply:
Aha, I was looking for something like this but I couldn't find it. Thanks!